File: app/src/main/java/jp/juggler/subwaytooter/api/entity/TootApplication.java
package jp.juggler.subwaytooter.api.entity;

import android.support.annotation.Nullable;

import org.json.JSONObject;

import jp.juggler.subwaytooter.util.LogCategory;
import jp.juggler.subwaytooter.util.Utils;

public class TootApplication {
	
	private static final LogCategory log = new LogCategory( "TootApplication" );
	
	public String name;
	public String website;
	
	@Nullable
	public static TootApplication parse( JSONObject src ){
		if( src == null ) return null;
		try{
			TootApplication dst = new TootApplication();
			dst.name = Utils.optStringX( src, "name" );
			dst.website = Utils.optStringX( src, "website" );
			return dst;
		}catch( Throwable ex ){
			log.trace( ex );
			log.e( ex, "TootApplication.parse failed." );
			return null;
		}
	}
}

File: app/src/main/java/jp/juggler/subwaytooter/api/entity/TootApplication.kt
package jp.juggler.subwaytooter.api.entity

import jp.juggler.subwaytooter.api.TootParser
import jp.juggler.util.JsonObject

class TootApplication(parser: TootParser, src: JsonObject) {

    val name: String?

    @Suppress("unused")
    private val website: String?

    //	val description : String?

    init {
        if (parser.serviceType == ServiceType.MISSKEY) {
            name = src.string("name")
            website = null
            //			description = src.string("description")
        } else {
            name = src.string("name")
            website = src.string("website")
            //			description = website
        }
    }
}
